extensions/toggl-track/src/helpers/allWorkspacesFetch.ts

```diff
@@ -1,13 +1,14 @@
 import { Workspace } from "../api";
 
-export function allWorkspacesFetch<T>(workSpaceFunction: (workspaceId: number) => Promise<T[] | null>) {
-  return async (workspaces: Workspace[]) => {
-    const data = await Promise.all(
-      workspaces.map(async (workspace) => {
-        const workspaceData = await workSpaceFunction(workspace.id);
-        return workspaceData || [];
-      }),
-    );
-    return data.flat();
-  };
+export async function allWorkspacesFetch<T>(
+  workSpaceFunction: (workspaceId: number) => Promise<T[] | null>,
+  workspaces: Workspace[],
+) {
+  const data = await Promise.all(
+    workspaces.map(async (workspace) => {
+      const workspaceData = await workSpaceFunction(workspace.id);
+      return workspaceData || [];
+    }),
+  );
+  return data.flat();
```

Comment:
So not returning a function was the fix in combination with moving the function outside of the hook?

Reply:
Great work man! This makes it easier to contribute with more features

Reply:
It was actually just moving the function outside of the hook. Return a function here was just unnecessary.